### PR TITLE
refactor: refactored deploy pipelines into single pipeline

### PR DESCRIPTION
**Pull request by simonstumpf**
## Overview

Please provide a brief description of what this pull request aims to achieve.

### Related Jira Story

KAN-153

### Backend Changes

- Summary of backend changes here

### Frontend Changes

- Summary of frontend changes here

### Printer Changes

- Summary of frontend changes here

## Definition of Done

- [ ] The feature must be fully tested and all specified requirements being met as outlined in the feature description.
- [ ] Unit and integrations test must be written for any business logic.
- [ ] Sufficient negative tests must be included to assess the robustness and error handling capabilities of the code.
- [ ] The pull request must be reviewed and approved by at least one codeowner of the modified files.
- [ ] All pipelines on the pull request succeed.
- [ ] Refactoring on the code has taken place.

## Screenshots

Include any relevant screenshots that will help in the review of this pull request.

## Additional Notes

Add any other notes about the pull request here, including any discussions, considerations, or dependencies.

---

Please review the above changes. Upon approval, please squash the commits following the guidelines before merging to the main branch.

### CODE REVIEW

File: .github/workflows/backend-push.yml [hunk 1]
```diff
@@ -1,8 +1,11 @@
 name: backend running on each commit
 
 on:
-  - push
-  - pull_request
+  push:
+    branches-ignore:
+      - main
+  pull_request:
```
Comment (Mcluky) [inline .github/workflows/backend-push.yml:7]: does that mean on all pull requests? Seems a bit strange that we don't further need to specify this

File: .github/workflows/deploy-stage.yml [hunk 2]
```diff
@@ -4,13 +4,16 @@ on:
   workflow_dispatch:
   push:
     branches:
-      - main
+      - KAN-153
```
Comment (Mcluky) [inline .github/workflows/deploy-stage.yml:7]: we probably want to change that back to main right?

File: .github/workflows/backend-push.yml [hunk 3]
```diff
@@ -1,8 +1,11 @@
 name: backend running on each commit
 
 on:
-  - push
-  - pull_request
+  push:
+    branches-ignore:
+      - main
```
Comment (Mcluky) [inline .github/workflows/backend-push.yml:6]: Should sonarcloud scan not also run on the main? With this change that would not run anymore on main right?

**Review by Mcluky — APPROVED** [review]

File: .github/workflows/backend-push.yml (hunk 1 above)
Comment (simonstumpf) [reply .github/workflows/backend-push.yml:7]: yeah its that simple. you cold do it also as an array (but then you couldn't configure the branches-ignore)

File: .github/workflows/deploy-stage.yml (hunk 2 above)
Comment (simonstumpf) [reply .github/workflows/deploy-stage.yml:7]: i changed it :)

File: .github/workflows/backend-push.yml (hunk 3 above)
Comment (simonstumpf) [reply .github/workflows/backend-push.yml:6]: yeah it should, but this is done during the stage deployment (also triggered main).

**Pull request merged**